src/ophyd_async/core/flyer.py

```diff
@@ -242,7 +246,9 @@ async def kickoff(self) -> None:
     async def _fly(self) -> None:
         await self._trigger_logic.start()
         # Wait for all detectors to have written up to a particular frame
-        await self._detector_group_logic.wait_for_index(self._last_frame - self._offset)
+        await self._detector_group_logic.wait_for_index(
+            self._last_frame - self._offset, timeout=self._timeout
+        )
```

Comment:
By the time this is called, the triggers have been sent to the detector, so I think `DEFAULT_TIMEOUT` is a reasonable time to wait for all the frames to be written
```suggestion
        await self._detector_group_logic.wait_for_index(
            self._last_frame - self._offset, timeout=DEFAULT_TIMEOUT
        )
```